src/builders/indexable-author-builder.php

```diff
@@ -72,8 +73,29 @@ public function __construct(
 	 * @param Indexable $indexable The indexable to format.
 	 *
 	 * @return Indexable The extended indexable.
+	 *
+	 * @throws Author_Not_Built_Exception When author is not built.
 	 */
 	public function build( $user_id, Indexable $indexable ) {
+		if ( $this->author_archive->are_disabled() ) {
+			throw Author_Not_Built_Exception::author_archives_are_disabled( $user_id );
+		}
+
+		if ( $this->author_archive->are_not_indexed() ) {
```

Comment:
I think we should still build indexables if the objects are noindexed. All publicly viewable objects are indexable, but may or may not request not to be indexed.